### ISSUE TITLE
error loading a specific type of string via wisper-sidekiq

### ISSUE THREAD

**Issue by sairam** [krisleech/wisper-sidekiq#30]
Hi, 

This is the minimal reproducable steps I could come up with on ruby 2.7.1 . the error comes when we use a string which is an integer with an underscore.

example event: `{'hello' => '1234_'}`

source: https://github.com/krisleech/wisper-sidekiq/blob/9f3e92526a4a9492ea9b06c3301890daf07808d4/lib/wisper/sidekiq.rb#L35

I understand this is a YAML serialization error because of the type of string and it automatically attempting to convert. Do you think any other encoding may help to avoid this problem of serialization of strings as integers?

```
# test.rb with ruby 2.7.1 
begin
  require "bundler/inline"
rescue LoadError => e
  $stderr.puts "Bundler version 1.10 or later is required. Please update your Bundler"
  raise e
end

gemfile(true) do
  source "https://rubygems.org"

  gem "rails"
  gem "wisper"
  gem "wisper-sidekiq"
end

class WisperListener
  def self.user_updated(event)
    puts event
  end
end


Wisper.subscribe(WisperListener, async: true)

class User
  include Wisper::Publisher

  def submit
    publish('user_updated', 'hello' => '1234_')
  end
end

User.new.submit
```

Output: 
<details>

```
Fetching gem metadata from https://rubygems.org/.............
Fetching gem metadata from https://rubygems.org/.
Resolving dependencies...
Using rake 13.0.1
Using concurrent-ruby 1.1.6
Using zeitwerk 2.3.1
Using thread_safe 0.3.6
Using builder 3.2.4
Using erubi 1.9.0
Using crass 1.0.6
Using mini_portile2 2.4.0
Using nio4r 2.5.2
Using minitest 5.14.1
Using websocket-extensions 0.1.5
Using mimemagic 0.3.5
Using rack 2.2.3
Using connection_pool 2.2.3
Using bundler 2.1.4
Using thor 1.0.1
Using redis 4.2.1
Using wisper 2.0.1
Using method_source 1.0.0
Using tzinfo 1.2.7
Using nokogiri 1.10.9
Using websocket-driver 0.7.2
Using i18n 1.8.3
Using rack-test 1.1.0
Using mini_mime 1.0.2
Using sprockets 4.0.2 (was 3.7.2)
Using marcel 0.3.3
Using activesupport 6.0.3.2
Using rack-protection 2.0.8.1
Using rails-dom-testing 2.0.3
Using loofah 2.6.0
Using activemodel 6.0.3.2
Using globalid 0.4.2
Using rails-html-sanitizer 1.3.0
Using sidekiq 6.0.7
Using activerecord 6.0.3.2
Using actionview 6.0.3.2
Using wisper-sidekiq 1.3.0
Using actionpack 6.0.3.2
Using activejob 6.0.3.2
Using mail 2.7.1
Using railties 6.0.3.2
Using activestorage 6.0.3.2
Using actionmailer 6.0.3.2
Using actionmailbox 6.0.3.2
Using actiontext 6.0.3.2
Using sprockets-rails 3.2.1
Using actioncable 6.0.3.2
Using rails 6.0.3.2
```

</details>

Error: 

<details>

```
Traceback (most recent call last):
	28: from test.rb:36:in `<main>'
	27: from test.rb:32:in `submit'
	26: from /home/user/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/wisper-2.0.1/lib/wisper/publisher.rb:42:in `broadcast'
	25: from /home/user/.rbenv/versions/2.7.1/lib/ruby/2.7.0/set.rb:328:in `each'
	24: from /home/user/.rbenv/versions/2.7.1/lib/ruby/2.7.0/set.rb:328:in `each_key'
	23: from /home/user/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/wisper-2.0.1/lib/wisper/publisher.rb:43:in `block in broadcast'
	22: from /home/user/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/wisper-2.0.1/lib/wisper/registration/object.rb:18:in `broadcast'
	21: from /home/user/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/wisper-sidekiq-1.3.0/lib/wisper/sidekiq.rb:35:in `broadcast'
	20: from /home/user/.rbenv/versions/2.7.1/lib/ruby/2.7.0/psych.rb:513:in `dump'
	19: from /home/user/.rbenv/versions/2.7.1/lib/ruby/2.7.0/psych/visitors/yaml_tree.rb:118:in `push'
	18: from /home/user/.rbenv/versions/2.7.1/lib/ruby/2.7.0/psych/visitors/yaml_tree.rb:136:in `accept'
	17: from /home/user/.rbenv/versions/2.7.1/lib/ruby/2.7.0/psych/visitors/yaml_tree.rb:351:in `visit_Array'
	16: from /home/user/.rbenv/versions/2.7.1/lib/ruby/2.7.0/psych/visitors/yaml_tree.rb:359:in `visit_Enumerator'
	15: from /home/user/.rbenv/versions/2.7.1/lib/ruby/2.7.0/psych/visitors/yaml_tree.rb:359:in `each'
	14: from /home/user/.rbenv/versions/2.7.1/lib/ruby/2.7.0/psych/visitors/yaml_tree.rb:359:in `block in visit_Enumerator'
	13: from /home/user/.rbenv/versions/2.7.1/lib/ruby/2.7.0/psych/visitors/yaml_tree.rb:136:in `accept'
	12: from /home/user/.rbenv/versions/2.7.1/lib/ruby/2.7.0/psych/visitors/yaml_tree.rb:351:in `visit_Array'
	11: from /home/user/.rbenv/versions/2.7.1/lib/ruby/2.7.0/psych/visitors/yaml_tree.rb:359:in `visit_Enumerator'
	10: from /home/user/.rbenv/versions/2.7.1/lib/ruby/2.7.0/psych/visitors/yaml_tree.rb:359:in `each'
	 9: from /home/user/.rbenv/versions/2.7.1/lib/ruby/2.7.0/psych/visitors/yaml_tree.rb:359:in `block in visit_Enumerator'
	 8: from /home/user/.rbenv/versions/2.7.1/lib/ruby/2.7.0/psych/visitors/yaml_tree.rb:136:in `accept'
	 7: from /home/user/.rbenv/versions/2.7.1/lib/ruby/2.7.0/psych/visitors/yaml_tree.rb:328:in `visit_Hash'
	 6: from /home/user/.rbenv/versions/2.7.1/lib/ruby/2.7.0/psych/visitors/yaml_tree.rb:328:in `each'
	 5: from /home/user/.rbenv/versions/2.7.1/lib/ruby/2.7.0/psych/visitors/yaml_tree.rb:330:in `block in visit_Hash'
	 4: from /home/user/.rbenv/versions/2.7.1/lib/ruby/2.7.0/psych/visitors/yaml_tree.rb:136:in `accept'
	 3: from /home/user/.rbenv/versions/2.7.1/lib/ruby/2.7.0/psych/visitors/yaml_tree.rb:279:in `visit_String'
	 2: from /home/user/.rbenv/versions/2.7.1/lib/ruby/2.7.0/psych/scalar_scanner.rb:95:in `tokenize'
	 1: from /home/user/.rbenv/versions/2.7.1/lib/ruby/2.7.0/psych/scalar_scanner.rb:104:in `parse_int'
/home/user/.rbenv/versions/2.7.1/lib/ruby/2.7.0/psych/scalar_scanner.rb:104:in `Integer': invalid value for Integer(): "1234_" (ArgumentError)
```

</details>

**Comment by sairam** [comment]
This is a regression that failed in ruby psych library - fixed in mainline less than 30 days ago -  https://github.com/ruby/psych/pull/438 . This should be available in the next patch version of ruby

**Comment by sairam** [comment]
this is solved with `gem 'psych','~3.1.0'`

**Comment by sairam** [comment]
adding the gem does not work though.
